mtags/src/main/scala-2/scala/meta/internal/pc/SemanticTokenProvider.scala

```diff
@@ -0,0 +1,425 @@
+
+      // Interpolation 
+      case _: Token.Interpolation.Id
+          |_: Token.Interpolation.SpliceStart
+             => getTypeId(SemanticTokenTypes.Keyword)
+      case _: Token.Interpolation.Start
+          |_: Token.Interpolation.Part
+          |_: Token.Interpolation.SpliceEnd
+          |_: Token.Interpolation.End
+             => getTypeId(SemanticTokenTypes.String)  // $ symbol
+
+      // Default
+      case _ => -1
+    }
+
+  }
+  def pickFromTraversed(tk: scala.meta.tokens.Token): Option[NodeInfo] = {
+
+    val buffer = ListBuffer.empty[NodeInfo]
+    for (node <- nodes) {
+      if (
+        node.pos.start == tk.pos.start &&
+        node.pos.end == tk.pos.end
+      ) buffer.++=(List(node))
+    }
+
+    buffer.toList.headOption
+  }
+
+  case class NodeInfo(
+      sym: Option[Symbol],
+      pos: scala.reflect.api.Position
+  )
+  object NodeInfo {
+    def apply(tree: Tree, pos: scala.reflect.api.Position): NodeInfo =
+      new NodeInfo(Some(tree.symbol), pos)
+
+    def apply(sym: Symbol, pos: scala.reflect.api.Position): NodeInfo =
+      new NodeInfo(Some(sym), pos)
+  }
+
+  /**
+   * was written in reference to PcDocumentHighlightProvider.
+   */
+  import cp._
+  object traverser {
+
+    /**
+     * gathers all nodes inside given tree.
+     * The nodes have symbol.
+     */
+    def traverse(
+        nodes: List[NodeInfo],
+        tree: cp.Tree
+    ): List[NodeInfo] = {
+
+      tree match {
+        /**
+         * All indentifiers such as:
+         * val a = <<b>>
+         */
+        case ident: cp.Ident if ident.pos.isRange =>
+          nodes :+ NodeInfo(ident, ident.pos)
+        /**
+         * Needed for type trees such as:
+         * type A = [<<b>>]
+         */
+        case tpe: cp.TypeTree if tpe.original != null && tpe.pos.isRange =>
+          tpe.original.children.foldLeft(
+            nodes :+ NodeInfo(tpe.original, typePos(tpe))
+          )(traverse(_, _))
+        /**
+         * All select statements such as:
+         * val a = hello.<<b>>
+         */
+        case sel: cp.Select if sel.pos.isRange =>
+          traverse(
+            nodes :+ NodeInfo(sel, sel.namePos),
+            sel.qualifier
+          )
+        /* all definitions:
+         * def <<foo>> = ???
+         * class <<Foo>> = ???
+         * etc.
+         */
+        case df: cp.MemberDef if df.pos.isRange =>
+          (annotationChildren(df) ++ df.children)
+            .foldLeft(
+              nodes :+ NodeInfo(df, df.namePos)
+            )(traverse(_, _))
+        /* Named parameters, since they don't show up in typed tree:
+         * foo(<<name>> = "abc")
+         * User(<<name>> = "abc")
+         * etc.
+         */
+        case appl: cp.Apply =>
+          val named = appl.args
+            .flatMap { arg =>
+              namedArgCache.get(arg.pos.start)
+            }
+            .collectFirst { case cp.AssignOrNamedArg(i @ cp.Ident(_), _) =>
+              NodeInfo(i, i.pos)
```

Comment:
Seems like ident itself doesn't have a symbol, but we can just find the parameter of the method.
```suggestion
              NodeInfo(appl.symbol.paramss.flatten.find(_.name == i.name), i.pos)
```